esphome/components/sdm220m/sdm220m.cpp

```diff
@@ -0,0 +1,113 @@
+#include "sdm220m.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace sdm220m {
+
+static const char *TAG = "sdm220m";
+
+static const uint8_t MODBUS_CMD_READ_IN_REGISTERS = 0x04;
+static const uint8_t MODBUS_REGISTER_COUNT = 74;  // 74 x 16-bit registers
+
+void SDM220M::on_modbus_data(const std::vector<uint8_t> &data) {
+  if (data.size() < MODBUS_REGISTER_COUNT * 2) {
+    ESP_LOGW(TAG, "Invalid size for SDM220M!");
+    return;
+  }
+
+  auto sdm220m_get_16bit = [&](size_t i) -> uint16_t {
+    return (uint16_t(data[i + 0]) << 8) | (uint16_t(data[i + 1]) << 0);
+  };
+  auto sdm220m_get_32bit = [&](size_t i) -> uint32_t {
+    return (uint32_t(sdm220m_get_16bit(i + 0)) << 16) | (uint32_t(sdm220m_get_16bit(i + 2)) << 0);
+  };
+
+  uint32_t temp;
+
+  temp = sdm220m_get_32bit(0);
+  float voltage = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(12);
+  float current = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(24);
+  float active_power = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(36);
+  float apparent_power = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(48);
+  float reactive_power = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(60);
+  float power_factor = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(72);
+  float phase_angle = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(140);
+  float frequency = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(144);
+  float import_active_energy = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(148);
+  float export_active_energy = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(152);
+  float import_reactive_energy = *((float *) &temp);
+
+  temp = sdm220m_get_32bit(156);
+  float export_reactive_energy = *((float *) &temp);
+
+  ESP_LOGD(TAG,
```

Comment:
This probably does not need to be a debug message.

```suggestion
  ESP_LOGV(TAG,
```